### ISSUE TITLE
render error callback not called on import path error

### ISSUE THREAD

**Issue by maxlath** [sass/node-sass#1592]
following the [removal of working directory-relative import](http://sass-lang.com/documentation/file.SASS_CHANGELOG.html#deprecations__must_read) (which I guess is the breaking change mentioned in #1474 ), compiling my old scss code rejected an error of the kind:

``` json
{
  "formatted": "Error: File to import not found or unreadable: ../bower_components/foundation/scss/foundation/components/alert-boxes\n       Parent style sheet: /home/maxlath/inventaire/client/app/modules/general/scss/foundation/_imports.scss\n        on line 29 of app/modules/general/scss/foundation/_imports.scss\n>> @import \"../bower_components/foundation/scss/foundation/components/alert-boxes\n   ^\n",
  "message": "File to import not found or unreadable: ../bower_components/foundation/scss/foundation/components/alert-boxes\nParent style sheet: /home/maxlath/inventaire/client/app/modules/general/scss/foundation/_imports.scss",
  "column": 1,
  "line": 29,
  "file": "/home/maxlath/inventaire/client/app/modules/general/scss/foundation/_imports.scss",
  "status": 1
}
```

But while I do get this error from the executable `node-sass/bin/node-sass`, I never get it from the module itself: [`binding.render`](https://github.com/sass/node-sass/blob/adb6166b5570e567fa31ea98d623fc0892be9bf5/lib/index.js#L350) is called, but neither the [`error`](https://github.com/sass/node-sass/blob/adb6166b5570e567fa31ea98d623fc0892be9bf5/lib/index.js#L270) nor the [`success`](https://github.com/sass/node-sass/blob/adb6166b5570e567fa31ea98d623fc0892be9bf5/lib/index.js#L278), letting the callback passed to render hanging. Any clue where this difference of behavior comes from?

**Comment by maxlath** [comment]
I made a [demo project of the error](https://github.com/maxlath/sass-fail-demo) to illustrated [a related issue in another project](https://github.com/brunch/brunch/issues/1402) but that could help here too

**Comment by xzyfer** [comment]
>following the removal of working directory-relative import 

That is the Sass changelog, not Node Sass.

> which I guess is the breaking change mentioned in #1474 

It is not. Please take a moment to read the post before [spreading misinformation](https://github.com/brunch/brunch/issues/1402) like this:

>A breaking change in node-sass made my build break, but apparently without properly rejecting the error. 

***

A simple reduce test case show the error is produced as expected

```sass
// index.scss
@import "no-such-file";
```
```
./node_modules/.bin/node-sass index.scss
```
```json
{
  "formatted": "Error: File to import not found or unreadable: no-such-file\n       Parent style sheet: /Users/michael/tmp/issue-1592/index.scss\n        on line 1 of index.scss\n>> @import \"no-such-file\"\n   ^\n",
  "message": "File to import not found or unreadable: no-such-file\nParent style sheet: /Users/michael/tmp/issue-1592/index.scss",
  "column": 1,
  "line": 1,
  "file": "/Users/michael/tmp/issue-1592/index.scss",
  "status": 1
}
```

***

This is clearly an issue with brunch or which package is call `node-sass`.

**Comment by maxlath** [comment]
@xzyfer my apologies for the mismatch on the breaking changes, I got to say I'm still confused on when the change on working directory import path happened then.

There is still something bothering me here:

* this test logs the expected error:
```javascript
var sass = require('node-sass')
var options = {
  file: 'whatever'
}
sass.render(options, function(err, result){
  console.log('err', err)
  console.log('result', result)
})
```

* but this case stays silent and keeps eating the CPU
```javascript
var sass = require('node-sass')
var options = {
  file: 'whatever',
  data: '@import \'../some/inexisting/file\';',
  includePaths: [ '.', 'app' ]
}
sass.render(options, function(err, result){
  console.log('err', err)
  console.log('result', result)
})
```

It seems it either has to do with the `../` path of the `@import` or the `.` in includePaths: if you remove either of those, you get your error message back.

**Comment by xzyfer** [comment]
Thanks for your understanding. Using both data and file is unsupported and
results in defined behaviour. This is likely causing the issue you're
describing.

**Comment by maxlath** [comment]
Sorry to insiste, but it doesn't look like the issue is that I'm passing both file and data. We can actually remove file and still get a case without any error being returned:
* this gets the right error
```js
var options = {
  data: '@import \'../some/inexisting/file\';',
  includePaths: [ 'app' ]
}
```
* this too
```js
var options = {
  data: '@import \'./some/inexisting/file\';',
  includePaths: [ '.', 'app' ]
}
```
* but not this
```js
var options = {
  data: '@import \'../some/inexisting/file\';',
  includePaths: [ '.', 'app' ]
}
```

**Comment by xzyfer** [comment]
>Sorry to insiste, but it doesn't look like the issue is that I'm passing both file and data. 

Sorry, to be clear, passing both data and file is probably why you're seeing the a different issue

> this one stays silent and keeps eating the CPU

**Comment by xzyfer** [comment]
I am able to reproduce a process hang with

```js
var sass = require('node-sass')
var options = {
  data: '@import "../does-not-exist";',
  includePaths: ['.']
}
sass.render(options, function(err, result){
  console.log('err', err)
  console.log('result', result)
})
```

**Comment by xzyfer** [comment]
After some digging it looks like the issue is with `.` being used a load path. Using `__dirname` or an absolute path instead works as expected.

**Comment by xzyfer** [comment]
I have confirmed this a bug in LibSass - https://github.com/sass/libsass/issues/2106

**Comment by michaelhogg** [comment]
The bug in LibSass was fixed in sass/libsass#2107, and the fix was released in [v3.4.0](https://github.com/sass/libsass/releases/tag/3.4.0).

The version of LibSass used by node-sass was [bumped to 3.4.0](https://github.com/sass/node-sass/commit/6532189) for [release v4.0.0](https://github.com/sass/node-sass/releases/tag/v4.0.0).

So I think this issue should be resolved in v4.0.0 onwards.
